### PR TITLE
Add kube-proxy e2e test for services with no endpoints

### PR DESCRIPTION
**Pull request by spencerhance**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Adds e2e test to verify that kube-proxy doesn't let in traffic from a loadbalancer to a service with no endpoints.  Different from https://github.com/kubernetes/kubernetes/pull/72561 which doesn't use an external LB as far as I can tell.

Added due to:
https://github.com/kubernetes/kubernetes/pull/72534

**Special notes for your reviewer**:
The plan is to cherry pick this to 1.11, 1.12, and 1.13 like as is being done with https://github.com/kubernetes/kubernetes/pull/72534.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/network/proxy.go [hunk 1]
```diff
@@ -254,6 +256,78 @@ var _ = SIGDescribe("Proxy", func() {
...
+					Ports: []v1.ServicePort{
+						v1.ServicePort{
+							Name: "portname1",
+							Port: svcPort,
+							TargetPort: intstr.FromInt(80),
+						},
+					},
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
```
Comment (neolit123) [inline test/e2e/network/proxy.go]: please annotate this error.
[1]

File: test/e2e/network/proxy.go [hunk 2]
```diff
@@ -254,6 +256,78 @@ var _ = SIGDescribe("Proxy", func() {
...
+			svc = jig.WaitForLoadBalancerOrFail(f.Namespace.Name, svcName, framework.LoadBalancerCreateTimeoutDefault)
+			nodeList := jig.GetNodes(framework.MaxNodesForEndpointsTests)
+			Expect(nodeList).NotTo(BeNil())
```
Comment (neolit123) [inline test/e2e/network/proxy.go]: [1]

File: test/e2e/network/proxy.go [hunk 3]
```diff
@@ -254,6 +256,78 @@ var _ = SIGDescribe("Proxy", func() {
...
+			nodesNames := jig.GetNodesNames(framework.MaxNodesForEndpointsTests)
+			if len(nodesNames) <= 0 {
+				framework.Failf("Expect at least 1 node, got: %v", nodesNames)
```
Comment (neolit123) [inline test/e2e/network/proxy.go]: `Expected`

File: test/e2e/network/proxy.go [hunk 4]
```diff
@@ -254,6 +256,78 @@ var _ = SIGDescribe("Proxy", func() {
...
+									Image: imageutils.GetE2EImage(imageutils.ServeHostname),
+									Ports: []v1.ContainerPort{{ContainerPort: svcPort, Name: "web"}},
+								},
+							},
+							HostNetwork: true,
+						},
+					},
+				},
+			})
+			Expect(err).NotTo(HaveOccurred())
```
Comment (neolit123) [inline test/e2e/network/proxy.go]: [1]

File: test/e2e/network/proxy.go [hunk 5]
```diff
@@ -254,6 +256,78 @@ var _ = SIGDescribe("Proxy", func() {
...
+
+			nodesSet := sets.NewString(nodesNames...)
+			svcExternalIP := svc.Status.LoadBalancer.Ingress[0].IP
+
+			// Make sure pod isn't reachable through the service with no pods
+			Expect(framework.TestHitNodesFromOutside(svcExternalIP, svcPort, 2*time.Minute, nodesSet)).To(HaveOccurred())
```
Comment (neolit123) [inline test/e2e/network/proxy.go]: [1]

**Review by neolit123 — COMMENTED** [review]
thanks for the PR

/approve
for the framework typos.

for e2e/network
@kubernetes/sig-network-pr-reviews 

please annotate the errors with proper descriptions.
thanks.

File: test/e2e/network/proxy.go (hunk 1 above)
Comment (spencerhance) [reply test/e2e/network/proxy.go]: Done

File: test/e2e/network/proxy.go (hunk 2 above)
Comment (spencerhance) [reply test/e2e/network/proxy.go]: Done

File: test/e2e/network/proxy.go (hunk 3 above)
Comment (spencerhance) [reply test/e2e/network/proxy.go]: Done

File: test/e2e/network/proxy.go (hunk 4 above)
Comment (spencerhance) [reply test/e2e/network/proxy.go]: Done

File: test/e2e/network/proxy.go (hunk 5 above)
Comment (spencerhance) [reply test/e2e/network/proxy.go]: Done

File: test/e2e/network/proxy.go (hunk 1 above)
Comment (neolit123) [reply test/e2e/network/proxy.go]: sorry i meant this:
https://onsi.github.io/gomega/
see "Annotating Assertions"

Comment (spencerhance) [reply test/e2e/network/proxy.go]: whoops, my bad.  Will do.

Comment (spencerhance) [reply test/e2e/network/proxy.go]: Updated

**Comment by neolit123** [comment]
thanks for update.
please squash the commits into one.

/priority important-longterm
/ok-to-test

**Comment by spencerhance** [comment]
@neolit123 thanks!

**Comment by spencerhance** [comment]
closing since it seems that https://github.com/kubernetes/kubernetes/pull/74394 updated an exisiting e2e test to be thorough enough.  Sorry for the noise.

**Pull request closed without merging**